src/org/jcodings/specific/CESU8Encoding.java

```diff
@@ -161,6 +160,14 @@ public int mbcToCode(byte[] bytes, int p, int end) {
     return c;
   }
 
+  private static int utf8Decode3ByteSequence(byte[] bytes, int p) {
+    return utf8Decode3ByteSequence(bytes, p, bytes[p]);
+  }
+
+  private static int utf8Decode3ByteSequence(byte[] bytes, int p, int c) {
+    return ((c & 0xF) << 12) | ((bytes[p + 1] & 0xff & 0x3f) << 6) | (bytes[p + 2] & 0xff & 0x3f);
```

Comment:
It was pointed out to me that 0x3f is not needed here as 0xff contains same bit pattern (sans 1 bit).  I am sure the compiler will figure this out so unless there is some cognitive benefit to seeing it?

Reply:
`0xff` is not needed, but `0x3f` is. I suppose `0xff` was introduced during conversion from C to Java to account for signedness. I just kept the `& 0xff` because it seems to be used everywhere in the JCodings implementation. Should I remove it in this PR?

Reply:
@djoooooe Ah yeah I said that backwards.   It probably just comes down to whether this is worth leaving in to make it simpler to compare.  I don't think there is harm in leaving it.  The only argument for removing it would be Java people studying wondering why we have an extra bit AND we don't need.

Reply:
FWIW this probably isn't too important because all accesses of unsigned byte values should move to the Java 8 `Byte.toUnsignedInt` anyway. If that means we do the unsigned conversion and then duplicate some work with an additional mask, I think it's still better than making other contributors guess at why one particular byte access doesn't have the unsigned masking.